lib/redis/namespace.rb

```diff
@@ -241,6 +241,7 @@ def initialize(namespace, options = {})
       @deprecations = !!options.fetch(:deprecations) do
                         ENV['REDIS_NAMESPACE_DEPRECATIONS']
                       end
+      @is_client_respond = @redis.respond_to?(:_client)
```

Comment:
I'd call this something like `@has_new_client_method` or `@responds_to_new_client_method`... or maybe just `@new_client_method`.